### PR TITLE
fixed mama code to work on raw data

### PR DESCRIPTION
**Pull request by manumanuk**
This change is [Reviewable](https://reviewable.io/reviews/waterloo-rocketry/canlib/33)

### CODE REVIEW

**Review by zx100x100 — CHANGES_REQUESTED** [review]
Looks like the get_radi_info function on line 630 hasn't been updated, that needs to happen

also just to confirm the switch here is from a floating point value representation to the raw ADC value? 

It would be much nicer if you can simply accept a single 16 bit integer instead of two 8 bit ones representing the high and low portions of the value. Inside the function you would then break the single value out into the two bytes needed for the CAN message.

Finally not sure if you did, but make sure you actually run the unit tests and make sure they pass

**Comment by manumanuk** [comment]
* Changed get_radi_info function
* The floating point was the raw ADC value, yes
* Changed build_radi_info function to accept 16 bit value (do you want me to change get_radi_info to output a 16 bit too?
* Unit tests pass

**Review by wendi-yu — COMMENTED** [review]
code-wise this looks fine to me, I'll let Zach rereview on the board specifics though

> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/canlib/33#-:-MeYWh2D-I9La4s8AiRz:b-c7duoc)* status: 0 of 4 files reviewed, all discussions resolved (waiting on @zx100x100)

File: can_common.c [hunk 1]
```diff
@@ -344,8 +343,8 @@ bool build_radi_info_msg(uint32_t timestamp,
     write_timestamp_3bytes(timestamp, output);
 
     output->data[3] = sensor_identifier;
-    output->data[4] = int_value;
-    output->data[5] = deci_value;
+    output->data[4] = (uint8_t) (adc_value >> 8); // Upper byte
+    output->data[5] = (uint8_t) (adc_value); // Lower byte
```
Comment (zx100x100) [inline can_common.c:347]: Adding a mask vs relying on the implicit truncation of casting it to a uint8_t would be nice

**Review by zx100x100 — APPROVED** [review]
Aside from the one comment on adding a mask looks good to me. if you can address that it would be appreciated, but I'm not going to set the review as request changes

**Review by BluCodeGH — APPROVED** [review]
> Reviewed 4 of 4 files at r3.
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/canlib/33)* status: all files reviewed, 1 unresolved discussion (waiting on @manumanuk)

**Pull request merged**